src/operations/indexes.ts

```diff
@@ -153,24 +153,23 @@ export class IndexesOperation extends AbstractOperation<Document> {
 export class CreateIndexesOperation extends CommandOperation<Document> {
   options: CreateIndexesOptions;
   collectionName: string;
-  onlyReturnNameOfCreatedIndex?: boolean;
+  singular?: boolean;
   indexes: IndexDescription[];
 
   constructor(
     parent: OperationParent,
     collectionName: string,
     indexes: IndexDescription[],
-    options?: CreateIndexesOptions
+    options?: CreateIndexesOptions,
+    singular?: boolean
   ) {
     super(parent, options);
 
     this.options = options ?? {};
+    this.singular = singular ?? false;
```

Comment:
Do you even need to track this anymore? It seems like the way to do this is to have this operation always return `string[]`, and then have `CreateIndexOperation` do:
```
 execute(server: Server, session: ClientSession, callback: Callback<string>): void {
   super.execute(server, session, (err, indexNames) => {
     if (err) return callback(err);
     callback(undefined, indexNames[0]);
   });
 }
```
Alternatively you could have `CreateIndexesOperation` return `string | string[]` and get rid of `CreateIndexOperation` altogether.

Reply:
updated, PTAL 🙏